kpatch-build/kpatch-elf.h

```diff
@@ -0,0 +1,186 @@
+/*
+ * kpatch-elf.h
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA,
+ * 02110-1301, USA.
+ */
+
+#ifndef _KPATCH_ELF_H_
+#define _KPATCH_ELF_H_
+
+#include <gelf.h>
+#include "list.h"
+
+#define ERROR(format, ...) \
+	error(1, 0, "ERROR: %s: %d: " format, __FUNCTION__, __LINE__, ##__VA_ARGS__)
```

Comment:
It would be really nice if kpatch-elf.c still had access to childobj so that ERROR() and other logging macros can still use it.  Maybe it could just be part of the "interface": when using log.h you have to define childobj and loglevel in your .c file.